pkg/k8shandler/visualization.go

```diff
@@ -138,7 +141,14 @@ func (clusterRequest *ClusterLoggingRequest) removeKibana() (err error) {
 
 func (clusterRequest *ClusterLoggingRequest) createOrUpdateKibanaDeployment() (err error) {
 
-	kibanaPodSpec := newKibanaPodSpec(clusterRequest.cluster, "kibana", "elasticsearch")
+	proxy := &configv1.Proxy{}
+	if err = clusterRequest.client.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, proxy); err != nil {
+		if !errors.IsNotFound(err) {
+			logrus.Debugf("kibana: Failed to get proxy: %v\n", err)
+		}
+	}
+
+	kibanaPodSpec := newKibanaPodSpec(clusterRequest.cluster, "kibana", "elasticsearch.openshift-logging.svc.cluster.local", proxy)
```

Comment:
this is a lot of hardcoding for the elasticsearch endpoint... can we build the name or move it into a const instead?